src/libraries/System.DirectoryServices.Protocols/src/System/DirectoryServices/Protocols/ldap/LdapConnection.Windows.cs

```diff
@@ -0,0 +1,64 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Globalization;
+using System.Net;
+using System.Collections;
+using System.ComponentModel;
+using System.Text;
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+using System.Xml;
+using System.Threading;
+using System.Security.Cryptography.X509Certificates;
+
+namespace System.DirectoryServices.Protocols
+{
+    public partial class LdapConnection
+    {
+        private void InternalInitConnectionHandle(string hostname)
+        {
+            // User wants to setup a connectionless session with server.
+            if (((LdapDirectoryIdentifier)_directoryIdentifier).Connectionless == true)
+            {
+                _ldapHandle = new ConnectionHandle(Wldap32.cldap_open(hostname, ((LdapDirectoryIdentifier)_directoryIdentifier).PortNumber), _needDispose);
+            }
+            else
+            {
+                _ldapHandle = new ConnectionHandle(Wldap32.ldap_init(hostname, ((LdapDirectoryIdentifier)_directoryIdentifier).PortNumber), _needDispose);
+            }
+        }
+
+        private int InternalConnectToServer()
+        {
+            // Connect explicitly to the server.
+            var timeout = new LDAP_TIMEVAL()
+            {
+                tv_sec = (int)(_connectionTimeOut.Ticks / TimeSpan.TicksPerSecond)
```

Comment:
If this is just moved code it might be OK, but truncation may not be the right answer here.  Either round-up or round may be better.  E.g. is (int)(1.8 * TicksPerSecond) 1 second, or 2?  Either answer is OK, except 1 tick turns into 0, which may be special.

Reply:
It is code that was just moved over:

https://github.com/dotnet/runtime/blob/2ae694d32944f163b3b222d6077ecb9801bafe81/src/libraries/System.DirectoryServices.Protocols/src/System/DirectoryServices/Protocols/ldap/LdapConnection.cs#L1032-L1036